### ISSUE TITLE
ingeststore crashes with "panic: send on closed channel"

### ISSUE THREAD

**Issue by promise6522** [oklog/oklog#94]
oklog version: 0.3.0
I tested on a single linux node with 2 ingeststore nodes(localhost:7659/localhost:8659) and 1 forward node behind a testsvc service (rate=500).

```
../oklog-0.3.0-linux-amd64 ingeststore -cluster tcp://192.168.1.251:8659 -api tcp://192.168.1.251:8650 -ingest.bulk tcp://192.168.1.251:8653 -ingest.durable tcp://192.168.1.251:8652 -ingest.fast tcp://192.168.1.251:8651 -peer 192.168.1.251:7659 -debug -store.segment-retain 36h
ts=2017-12-25T05:05:42.535366573Z level=info cluster_bind=192.168.1.251:8659
ts=2017-12-25T05:05:42.53573753Z level=info fast=tcp://192.168.1.251:8651
ts=2017-12-25T05:05:42.5358706Z level=info durable=tcp://192.168.1.251:8652
ts=2017-12-25T05:05:42.535963723Z level=info bulk=tcp://192.168.1.251:8653
ts=2017-12-25T05:05:42.536045331Z level=info API=tcp://192.168.1.251:8650
ts=2017-12-25T05:05:42.536488814Z level=info ingest_path=data/ingest
ts=2017-12-25T05:05:42.536744071Z level=info store_path=data/store
ts=2017-12-25T05:05:42.536878988Z level=debug component=cluster bind_addr=192.168.1.251 bind_port=8659 ParseIP=192.168.1.251
ts=2017-12-25T05:05:42.540306902Z level=debug component=cluster received=NotifyJoin node=e1f99b32-149f-4c80-9563-3b7f69acb992 addr=192.168.1.251:8659
ts=2017-12-25T05:05:42.545577756Z level=debug component=cluster received=NotifyJoin node=d45f3766-b01d-4642-8649-b8c06e140209 addr=192.168.1.251:7659
ts=2017-12-25T05:05:42.54580084Z level=debug component=cluster Join=1
**panic: send on closed channel

goroutine 1592 [running]:
github.com/oklog/oklog/pkg/stream.readOnce(0xd624c0, 0xc42410b580, 0xc42420f120, 0xc4223fa0e0, 0x12, 0xc4203aeea0, 0x0, 0x0)
        /Users/pbourgon/src/github.com/oklog/oklog/pkg/stream/stream.go:132 +0x2b0
github.com/oklog/oklog/pkg/stream.readUntilCanceled(0xd624c0, 0xc42410b580, 0xc42420f120, 0xc4223fa0e0, 0x12, 0xc4203aeea0, 0xa88480)
        /Users/pbourgon/src/github.com/oklog/oklog/pkg/stream/stream.go:115 +0x78
created by github.com/oklog/oklog/pkg/stream.updateActive
        /Users/pbourgon/src/github.com/oklog/oklog/pkg/stream/stream.go:94 +0x201**
```

Thank you for this elegant project, i'm looking forward to deploy oklog in production environments.

**Comment by peterbourgon** [comment]
Yep, this is currently being fixed in #91, I just have to work out a problem with a test.

**Comment by peterbourgon** [comment]
Fixed.
